src/overlays/actors/ovl_Bg_Hidan_Firewall/z_bg_hidan_firewall.h

```diff
@@ -4,12 +4,29 @@
 #include <ultra64.h>
 #include <global.h>
 
-typedef struct
+struct BgHidanFirewall;
+
+typedef void FirewallFunc(struct BgHidanFirewall* this, GlobalContext* globalCtx);
+
+typedef struct BgHidanFirewall
 {
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x54];
+    /* 0x0000 */ Actor                 actor;
+    /* 0x014C */ FirewallFunc*         actionFunc;
+    /* 0x0150 */ s16                   unk_150;
+    /* 0x0154 */ ColliderCylinderMain  collider;
 } BgHidanFirewall; // size = 0x01A0
 
-extern const ActorInit Bg_Hidan_Firewall_InitVars;
+void BgHidanFirewall_Init(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Destroy(BgHidanFirewall* this, GlobalContext* globalCtx);
+s32  BgHidanFirewall_Proximity(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Wait(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Countdown(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Erupt(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Collide(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_ColliderFollowPlayer(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Update(BgHidanFirewall* this, GlobalContext* globalCtx);
+void BgHidanFirewall_Draw(BgHidanFirewall* this, GlobalContext* globalCtx);
+
+extern u32 D_0600DA80;
```

Comment:
we've been keeping prototypes and externs in the c file and we should try and keep that consistent across files

Reply:
you can also make the type Gfx since we know its a display list by its usage